pkg/kubectl/cmd/set/set_selector.go

```diff
@@ -0,0 +1,280 @@
+	Mapper      meta.RESTMapper
+	Typer       runtime.ObjectTyper
+	Infos       []*resource.Info
+	Encoder     runtime.Encoder
+	Out         io.Writer
+	Err         io.Writer
+	Filenames   []string
+	Recursive   bool
+	ShortOutput bool
+	Overwrite   bool
+	All         bool
+	Record      bool
+	ChangeCause string
+	Local       bool
+	DryRun      bool
+	Cmd         *cobra.Command
+
+	PrintObject             func(cmd *cobra.Command, mapper meta.RESTMapper, obj runtime.Object, out io.Writer) error
+	UpdateSelectorForObject func(obj runtime.Object, fn func(*map[string]string, *unversioned.LabelSelector) error) error
+	Resources               []string
+	Selector                *unversioned.LabelSelector
+}
+
+var (
+	selectorLong = dedent.Dedent(`
+		Update the selectors on a resource.
+
+		A selector must begin with a letter or number, and may contain letters, numbers, hyphens, dots, and underscores, up to %[1]d characters.
+		If --overwrite is true, then existing selectors can be overwritten, otherwise attempting to overwrite a selector will result in an error.
+		If --resource-version is specified, then updates will use this resource version, otherwise the existing resource-version will be used.`)
+	selectorExample = dedent.Dedent(`
+		# Update rc 'foo' with the selector 'unhealthy' and the value 'true'.
+		kubectl set selector rc foo unhealthy=true
+
+		# Update all deployments in the namespace
+		kubectl set selector deployment --all 'environment in (production, qa)'
+
+		# Update a rs identified by the type and name in "rs.json"
+		kubectl set selector -f rs.json 'environment notin (frontend)'
+
+		# Update service 'foo' only if the resource is unchanged from version 1.
+		kubectl set selector service foo status=unhealthy --resource-version=1`)
+)
+
+// NewCmdSelector is the "set selector" command.
+func NewCmdSelector(f *cmdutil.Factory, out io.Writer) *cobra.Command {
+	options := &SelectorOptions{
+		Out: out,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "selector [--overwrite] (-f FILENAME | TYPE NAME) EXPRESSIONS [--resource-version=version]",
+		Short:   "Update the selectors on a resource",
+		Long:    fmt.Sprintf(selectorLong),
+		Example: selectorExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(options.Complete(f, cmd, args))
+			cmdutil.CheckErr(options.Validate())
+			cmdutil.CheckErr(options.Run())
+		},
+	}
+	cmdutil.AddPrinterFlags(cmd)
+	cmd.Flags().Bool("all", false, "Select all resources in the namespace of the specified resource types")
+	cmd.Flags().String("resource-version", "", "If non-empty, the selectors update will only succeed if this is the current resource-version for the object. Only valid when specifying a single resource.")
+	usage := "Filename, directory, or URL to a file identifying the resource to update the selectors"
+	cmd.Flags().Bool("overwrite", false, "If true, allow selectors to be overwritten, otherwise reject selector updates that overwrite existing selectors.")
```

Comment:
I think we're almost always going to be overwriting, I don't see a need for this flag.